Performance rewrite (#1258)
OctopusDeploy/docs: docs/administration/managing-infrastructure/performance/create-sanitized-database-backup.md

The easiest way to do this is to use SQL Server Management Studio, right-click on the database, select **{{Tasks, Backup}}** and follow the wizard to create a full backup.  

You can also run this script to create the backup.

```sql
BACKUP DATABASE [OctopusDeploy]
  TO DISK = '\\SomeServer\SomeDrive\OctopusDeploy.bak'
      WITH FORMAT;
```

2. Restore the backup to a different server.

That new database is the database we are going to sanitize.  The easiest way to do this is to use SQL Server Management Studio, right-click on **Databases**, select **Restore Database** and follow the wizard.  We recommend naming it `OctopusDeploy_Sanitized` to make it clear as to the intention.

You can also run this script to restore the database.

```sql
USE [master]
RESTORE DATABASE [OctopusDeploy_Sanitized] 
    FROM  DISK = N'\\SomeServer\SomeDrive\OctopusDeploy.bak' WITH  FILE = 2,  
        MOVE N'OctopusDeploy' TO  N'YOUR_DATA_DIRECTORY\OctopusDeploy_Sanitized.mdf',  
        MOVE N'OctopusDeploy_log' TO N'YOUR_DATA_DIRECTORY\OctopusDeploy_Sanitized_log.ldf',  
    NOUNLOAD,  
    STATS = 5
GO
```

3. Disable deployment targets, project triggers, workers, and more.

> Run the following script to disable as much as possible on that database.  In the next step, you'll create an Octopus Deploy instance.  This script will prevent anything from running in the background that could potentially deploy or change anything in production.

Reviewer: ```suggestion
Run the following T-SQL script to disable as much as possible on that database.  In the next step, you'll create an Octopus Deploy instance.  This script will prevent anything from running in the background that could potentially deploy or change anything in production.
```

```sql
Use [OctopusDeploy_Sanitized]
go
DELETE FROM OctopusServerNode
IF EXISTS (SELECT null FROM sys.tables WHERE name = 'OctopusServerNodeStatus')
    DELETE FROM OctopusServerNodeStatus
UPDATE Subscription SET IsDisabled = 1
UPDATE ProjectTrigger SET IsDisabled = 1
UPDATE Machine SET IsDisabled = 1
IF EXISTS (SELECT null FROM sys.tables WHERE name = 'Worker')
    UPDATE Worker SET IsDisabled = 1
DELETE FROM ExtensionConfiguration WHERE Id in ('authentication-octopusid', 'jira-integration')
```

4. On a new server, install Octopus Deploy.

The installed version of Octopus Deploy should be the same version of Octopus Deploy you are running in production.  After installing Octopus Deploy, the Octopus Manager will appear.  You can close that and instead run these scripts to create the Octopus Deploy instance.

:::hint
Remember to run these scripts as **Administrator**.
:::

> ```

Reviewer: ```suggestion
```powershell
```

Set-Location "C:\Program Files\Octopus Deploy\Octopus"

.\Octopus.Server.exe create-instance --instance "Octopus" --config "C:\Octopus\OctopusServer.config" --serverNodeName "Sanitized"
.\Octopus.Server.exe database --instance "Octopus" --connectionString "Data Source=YOURSERVER;Initial Catalog=OctopusDeploy_Sanitized;Integrated Security=False;User ID=YOURUSER;Password=YOURPASSWORD"
```

:::hint
When you run the above commands, you will get a warning about being unable to decrypt the database.  You can ignore that.
:::

5. Sanitize the database.

> This command will clean out all sensitive variables and PII data and generate a new master key on the database.  **DO NOT** run this on your production instance.  You will have a bad day after this command finishes.

Reviewer: ```suggestion
This command will clean out all sensitive variables and PII data and generate a new master key on the database.  

:::warning
**DO NOT** run this on the database of your production instance.  Restoring any data lost after this command has finished executing is only possible using a full database backup along with the associated Master Key.
:::
```

> ```

Reviewer: ```suggestion
```powershell
```

Set-Location "C:\Program Files\Octopus Deploy\Octopus"

.\Octopus.Server.exe lost-master-key --instance "Octopus" --iReallyWantToResetAllMySensitiveData --upgradeDatabase --scrubPii --iHaveBackedUpMyDatabase
```

6. Create a new database backup.

> Create a new backup for OctopusDeploy_Sanitized.  That backup is the backup you will upload to us.

Reviewer: ```suggestion
Create a new backup for the `OctopusDeploy_Sanitized` database.  This is the backup you will upload to Octopus.
```
